File: doc-examples/example-kotlin/src/main/kotlin/io/micronaut/gcp/pubsub/subscriber/ContentTypeSubscriber.kt
/*
 * Copyright 2017-2020 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.gcp.pubsub.subscriber
//tag::imports[]
import com.google.pubsub.v1.PubsubMessage
import io.micronaut.gcp.pubsub.annotation.MessageId
import io.micronaut.gcp.pubsub.annotation.PubSubListener
import io.micronaut.gcp.pubsub.annotation.Subscription
import io.micronaut.gcp.pubsub.support.Animal
// end::imports[]

// tag::clazz[]
@PubSubListener
class ContentTypeSubscriber {
	@Subscription("raw-subscription")
	fun	receiveRaw(data: ByteArray, @MessageId id: String) { // <1>
	}

	@Subscription("native-subscription")
	fun  receiveNative(message: PubsubMessage) { // <2>
	}

	@Subscription("animals")
	fun receivePojo(animal: Animal, @MessageId id: String) { // <3>
	}

	@Subscription(value = "animals-legacy", contentType = "application/xml")
	fun  receiveXML(animal: Animal, @MessageId id: String) { // <4>
	}
}
// end::clazz[]

File: test-suite-python/src/main/python/micronaut/gcp/pubsub/subscriber/ContentTypeSubscriber.py
from typing import Annotated

# tag::imports[]
from com.google.pubsub.v1 import PubsubMessage
from micronaut.context.annotation import Requires
from micronaut.gcp.pubsub.annotation import MessageId, PubSubListener, Subscription
from micronaut.gcp.pubsub.support.Animal import Animal

from .MessageProcessor import MessageProcessor
# end::imports[]


@Requires(property="spec.name", value="ContentTypeSubscriberTest")
# tag::clazz[]
@PubSubListener
class ContentTypeSubscriber:

    def __init__(self, message_processor: MessageProcessor):
        self.message_processor = message_processor

    @Subscription("raw-subscription")  # <1>
    def receive_raw(self, data: bytes, id: Annotated[str, MessageId]) -> None:
        self.message_processor.handle_byte_array_message(data).block()

    @Subscription("native-subscription")  # <2>
    def receive_native(self, message: PubsubMessage) -> None:
        self.message_processor.handle_pub_sub_message(message).block()

    @Subscription("animals")  # <3>
    def receive_pojo(self, animal: Animal, id: Annotated[str, MessageId]) -> None:
        self.message_processor.handle_animal_message(animal).block()

    @Subscription(value="animals-legacy", contentType="application/xml")  # <4>
    def receive_xml(self, animal: Animal, id: Annotated[str, MessageId]) -> None:
        self.message_processor.handle_animal_message(animal).block()
# end::clazz[]
